WcaOnRails/app/controllers/api/v0/records_controller.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+class Api::V0::RecordsController < Api::V0::ApiController
+  def regionCondition(countrySource)
+    if params[:regionId]
+      region = params[:regionId]
+
+      if /^(world)?$/ =~ region
+        ''
+      elsif /^_/ =~ region
+        return "AND continentId = \'#{region}\'"
+      else
+        if countrySource
+          countrySource += '.'
+        end
+
+        return "AND #{countrySource}countryId = \'#{region}\'"
```

Comment:
I like this pattern for the dot handling:

`return "AND #{[countrySource, 'countryId'].compact.join('.')} = \'#{region}\'"`